core/lib/src/fairing/mod.rs

```diff
@@ -453,7 +453,7 @@ pub trait Fairing: Send + Sync + 'static {
     ///
     /// The default implementation of this method does nothing.
     #[allow(unused_variables)]
-    async fn on_response<'a>(&'a self, req: &'a Request<'_>, res: &'a mut Response<'_>) {}
+    async fn on_response(&self, req: &Request<'_>, res: &mut Response<'_>) {}
```

Comment:
```suggestion
    async fn on_response<'r>(&self, req: &'r Request<'_>, res: &mut Response<'r>) {}
```

I think we want this, so that the `Response` can borrow from the `Request`.

Reply:
Awesome feedback, that's a great lesson in niche lifetime design.